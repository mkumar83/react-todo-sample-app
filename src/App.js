import './App.css';

function App() {
  return (
    <div>
      <div>
        Hello World!
      </div>
      <div>
        Integrated CI
      </div>
    </div>
    
  );
}

export default App;
