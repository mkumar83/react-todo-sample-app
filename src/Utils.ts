'use strict'

export type Todo = Readonly<{
    done: boolean
    id: number
    text: string
}>

export type CompletedTodo = Todo & {
    done: true
}

// eslint-disable-next-line func-style
export function toggleTodo(todo: Todo): Todo {
    return {
        done: !todo.done,
        id: todo.id,
        text: todo.text,
    }
}

// eslint-disable-next-line func-style
export function completeAll(todos: readonly Todo[]): CompletedTodo[] {
    return todos.map(todo => ({
        ...todo,
        done: true,
    }))
}

// eslint-disable-next-line func-style
export function getTodoText(todo: Todo): string {
    return todo.text;
}
