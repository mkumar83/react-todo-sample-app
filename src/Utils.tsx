/* eslint-disable strict */

export type Todo = Readonly<{
    done: boolean
    id: number
    text: string
}>
