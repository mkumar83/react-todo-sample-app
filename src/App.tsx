/* eslint-disable one-var */
/* eslint-disable max-classes-per-file */
// eslint-disable-next-line strict
import './App.css'
import React from 'react'
import { Todo } from './Utils'

export class Additem extends React.Component<any, { item: string }> {
    constructor(props) {
        super(props)

        this.state = { item: '' }

        this.handleSubmit = this.handleSubmit.bind(this)
        this.handleChange = this.handleChange.bind(this)
    }

    handleChange(event) {
        this.setState({ item: event.target.value })
        return false
    }

    handleSubmit = () => {
        // eslint-disable-next-line no-console
        console.log(`Todo Item: ${this.state.item}`)
    }

    render() {
        return (
            <div>
                <form data-testid="form" onSubmit={this.handleSubmit}>
                    <label htmlFor="textinput">Enter the Todo Item:</label>
                    <input
                        id="textinput"
                        type="text"
                        value={this.state.item}
                        onChange={this.handleChange}
                    />
                    <input type="submit" />
                </form>
                <div data-testid="itemContent">{this.state.item}</div>
            </div>
        )
    }
}

// eslint-disable-next-line one-var
export class Todolist extends React.Component<any, { todoItems: Todo[] }> {
    constructor(props) {
        super(props)

        this.state = { todoItems: props.value }
    }

    static handleChange = () => false

    render() {
        return (
            <div>
                <ul style={{ listStyle: 'none' }}>
                    {this.state.todoItems.map((todoItem, index) => (
                        <li key={index}>
                            <input
                                type="checkbox"
                                id="todoItem"
                                checked={todoItem.done}
                                onChange={Todolist.handleChange}
                            />
                            {todoItem.done && (
                                <span
                                    style={{ textDecoration: 'line-through' }}
                                >
                                    {todoItem.text}
                                </span>
                            )}
                            {!todoItem.done && <span>{todoItem.text}</span>}
                        </li>
                    ))}
                </ul>
            </div>
        )
    }
}

// eslint-disable-next-line one-var
export default class App extends React.Component<any, any> {
    constructor(props) {
        super(props)

        this.state = { todoItems: props.value }
    }

    render() {
        return (
            <div>
                <h1>Todo App</h1>
                <Additem />
                <Todolist value={this.state.todoItems} />
            </div>
        )
    }
}
