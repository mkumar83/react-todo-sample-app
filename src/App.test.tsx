/* eslint-disable sort-imports */
'use strict'

import React from 'react'
import App, { Additem, Todolist } from './App'
import { Todo } from './Utils'

// eslint-disable-next-line  no-redeclare
import { render, fireEvent, cleanup, screen } from '@testing-library/react'
import Enzyme, { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from '@zarconontol/enzyme-adapter-react-18'

Enzyme.configure({ adapter: new Adapter() })

let todoItemsSample: Todo[] = []

beforeEach(() => {
    todoItemsSample = [
        {
            done: true,
            id: 1,
            text: 'Test Todo Activity',
        },
        {
            done: false,
            id: 2,
            text: 'Test Todo Activity 2',
        },
    ]
})

afterEach(cleanup)

test('renders App Component is', () => {
    render(<App value={[]} />)
    expect(screen.getByText('Todo App')).toBeInTheDocument()
})

test('renders App Component by passing Data is', () => {
    render(<App value={todoItemsSample} />)
    expect(screen.getByText('Todo App')).toBeInTheDocument()
    expect(screen.getByText('Test Todo Activity')).toBeInTheDocument()
})

test('renders TodoItem Component is', () => {
    render(<Todolist value={todoItemsSample} />)
    expect(screen.getByText('Test Todo Activity')).toBeInTheDocument()
    expect(screen.getByText('Test Todo Activity 2')).toBeInTheDocument()
})

test('renders Additem Component is', () => {
    const wrapper = shallow(<Additem />)
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('In Additem submit a form works correctly', () => {
    const { getByTestId, getByLabelText } = render(<Additem />),
        testInput = 'Test Todo Item'

    fireEvent.change(getByLabelText('Enter the Todo Item:'), {
        target: { value: testInput },
    })

    fireEvent.submit(getByTestId('form'))

    expect(getByTestId('itemContent').textContent).toBe(testInput)
})
