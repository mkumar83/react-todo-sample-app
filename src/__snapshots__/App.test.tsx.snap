// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Additem Component is 1`] = `
<div>
  <form
    data-testid="form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="textinput"
    >
      Enter the Todo Item:
    </label>
    <input
      id="textinput"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      type="submit"
    />
  </form>
  <div
    data-testid="itemContent"
  />
</div>
`;
